ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/TransactionReaper.java

```diff
@@ -725,18 +725,7 @@ private final void shutdown(boolean waitForTransactions)
                 notifyAll();
             }
 
-            /*
-                * Wait for all of the transactions to
-                * terminate normally.
-                */
-            while (!_reaperElements.isEmpty()) {
-                
-                try {
-                    this.wait(1000);
-                }
-                catch (final Exception ex) {
-                }
-            }
+            waitForAllTxnsToTerminate();
```

Comment:
Please can you clarify why this was moved out of the shutdown method?

Reply:
Sorry, I was not clear. I more am speaking to why it has to move outside of shutdown?

Reply:
To offer a public API to monitor what the Transaction Reaper is doing

Reply:
Thank you. Please feel free to resolve this thread.

Reply:
Thanks Tom